### PR TITLE
Adding eraser functionality to the Sketch Page

### PR DESCRIPTION
**Pull request by KanishkaJain30**
- Used destination-out property for an eraser that will remove those drawings where new drawings overlap the existing one
- Used source-over property for other tools that display the source image over the destination image.
- Users can also change the stroke-width of the eraser.
Resolves #708 

https://user-images.githubusercontent.com/52410283/111444937-e67c0700-8730-11eb-85f8-d5caba35c7d6.mp4

### CODE REVIEW

**Comment by anushbhatia** [comment]
Resolve the conflict.

File: src/pages/Sketch/components/Canvas.js [hunk 1]
```diff
@@ -211,8 +220,11 @@ function Canvas() {
     }
 
     function squareMove(point) {
+        context.globalCompositeOperation="source-over"; //Used source-over property for other tools that display the source image over the destination image.
         context.putImageData(typeState, 0, 0);
         context.beginPath();
+        console.log(downPoint);
+        console.log(point);
```
Comment (iamdarshshah) [inline src/pages/Sketch/components/Canvas.js:227]:
```suggestion
```
Avoid adding console logs while committing the code. Please remove them.

File: src/pages/Sketch/components/Canvas.js [hunk 2]
```diff
@@ -37,6 +37,7 @@ function Canvas() {
     /* ----- Canvas State ----- */
     const [isDrawing, setIsDrawing] = useState(false);
     const [type, setType] = useState("pen");
+    //const [type, setType] = useState("eraser");
```
Comment (iamdarshshah) [inline src/pages/Sketch/components/Canvas.js:40]:
```suggestion
```
Remove this, if it's not necessary.

File: src/pages/Sketch/components/Canvas.js [hunk 3]
```diff
@@ -109,11 +110,12 @@ function Canvas() {
         context.lineCap = 'round';
         context.lineWidth = width;
 
-        if(type === 'pen') {
+        if(type === 'pen'|| type==='eraser') { //if type is pen then logicDown function is called
```
Comment (iamdarshshah) [inline src/pages/Sketch/components/Canvas.js:113]:
```suggestion
        //if type is pen then logicDown function is called
        if(['pen', 'eraser'].includes(type)) { 
```
DO this instead. Please note: it's better to add comments prior to the actual logic. (update it where ever required.)

File: src/pages/Sketch/components/Canvas.js [hunk 4]
```diff
@@ -109,11 +110,12 @@ function Canvas() {
...
             logicDown(point);
         } else if(type === 'line' || type === 'square' || type === 'circle' || type === 'triangle' || type === 'arrow' || type === 'diamond') {
             setTypeState(context.getImageData(0, 0, canvasWidth, canvasHeight));
             logicDown(point);
+            //console.log(point);
```
Comment (iamdarshshah) [inline src/pages/Sketch/components/Canvas.js:118]:
```suggestion
```
Remove this as well.

**Comment by smaranjitghose** [comment]
New changes are not incorporated

**Pull request closed without merging**